ss/test/storage_test_suite.go

```diff
@@ -369,6 +369,34 @@ func (s *StorageTestSuite) TestDatabaseIteratorMultiVersion() {
 	s.Require().NoError(itr.Error())
 }
 
+// Tests bug where iterator loops continuously
+func (s *StorageTestSuite) TestDatabaseIteratorLooping() {
```

Comment:
We should add a test for reverse iteration as well

Reply:
Yea there are a lot more unit tests that are going to be pushed